primitives/runtime/src/generic/unchecked_extrinsic.rs

```diff
@@ -48,6 +49,25 @@ where
 	pub function: Call,
 }
 
+/// Manual [`TypeInfo`] implementation because of custom encoding. The data is a valid encoded
+/// `Vec<u8>`, but requires some logic to extract the signature and payload.
+///
+/// See [`UncheckedExtrinsic::encode`] and [`UncheckedExtrinsic::decode`].
+impl<Address, Call, Signature, Extra> TypeInfo
+	for UncheckedExtrinsic<Address, Call, Signature, Extra>
+where
+	Extra: SignedExtension + TypeInfo
+{
+	type Identity = UncheckedExtrinsic<(), (), (), Extra>;
+
+	fn type_info() -> Type {
+		Type::builder()
+			.path(Path::new("UncheckedExtrinsic", module_path!()))
+			.docs(&["UncheckedExtrinsic raw bytes, requires custom decoding routine"])
+			.composite(Fields::unnamed().field(|f| f.ty::<Vec<u8>>()))
```

Comment:
I see that the type info doesn't give that much information about this type,
It kind of encode like `(u32, SpecialMixOfSignedAndVersionAndSignature, Call)`.
If it is not needed to give more information about this type then I'm fine.

Reply:
Yeah it's because of the custom encoding logic which can't be represented in the type information. I did think about having something like you describe, but decided it's better to keep it opaque. 

So then it will be up to consumers e.g. `subxt` whether to define their own `UncheckedExtrinsic` type and encoding/decoding impls, or could even import `sp-primitives` and reuse the types here. This gives maximum flexibility, but the default of "just decode as a `Vec<u8>`" will still work.